### DISCUSSION TITLE
[Requesting feedback] New horizontal asset DAGs in the UI

### DISCUSSION THREAD

**Discussion by braunjj** [dagster-io/dagster#15426] [Ideas]
Asset Graphs in the UI tend to explode horizontally as the number of assets increase. This often creates odd-looking DAGs that stretch out super wide. Users are forced to pan across a really wide graph and it's difficult to get the full graph in view.

One suggestion we've heard to resolve this is to render the graph horizontally left-to-right instead of vertically. The hypothesis is that this new horizontal layout allows us to render that graph more compactly and this might feel more natural to navigate. 

With that said, there doesn't seem to be a clear consensus around vertical or horizontal DAGs in data tools (we find examples of both across data tools today), and this may end up simply being a user preference in the end.

We're interested in hearing from the community on this topic. We've built an experimental version of the horizontal asset DAGs that you can try out now. This new layout only impacts Asset Graphs (The classic op graph is always still rendered vertically).

### Demo
https://www.loom.com/share/9713f899dbe6484393d9404825405a8c

### Screenshots
This shows the same graph rendered vertically vs. horizontally 
**Vertical**:
![image](https://github.com/dagster-io/dagster/assets/2798333/e749a7bc-353d-4454-93d2-e979a009a415)

**Horizontal (new):**
![image](https://github.com/dagster-io/dagster/assets/2798333/c32e994e-f966-4498-9341-54841566f3e4)

**Pros of this approach:**
- In most cases the horizontal view fits the viewport better and is more compact. This means less panning and zooming.
- The asset graph layout is more similar to other data tools like dbt.
- Asset graphs are now even more visually distinct from op graphs

**Cons of this approach:**
- Throughout the UI horizontal views tend to represent elapsed time (the timeline view and the run gantt chart). This breaks that convention.
- It might be awkward and confusing that asset graphs and op graphs are rendered differently
- The word "Upstream" is a bit harder to grok when "up" = "left"

### How to enable the Horizontal Asset DAGs
You can try the new horizontal asset graph layout by enabling the `Experimental horizontal asset DAGs` feature flag in your user settings. Give it a try and let us know what you think. 
- Do you find the horizontal view more intuitive?
- Should asset graphs be rendered left-to-right by default moving forward?
- Anything else we should be considering here?

cc: @sryza @bengotow

**Comment by PedramNavid** [comment]
Very much in favor of horizontal as a default. Perhaps a button to switch between horizontal/vertical for cases where horizontal doesn't work as well could be helpful. Here's some real-world examples, I find Horizontal much more intuitive. 


<img width="1682" alt="image" src="https://github.com/dagster-io/dagster/assets/1045990/41013882-56d7-4891-9232-58e80b36735c">
<img width="1232" alt="image" src="https://github.com/dagster-io/dagster/assets/1045990/8f3a85ad-3f6c-4be9-a5a4-a026c0e3d70b">

**Reply by toddy86** [reply]
Plus one on this. @braunjj Is there a way to toggle back to a vertical view (I’m on Dagster 1.6.4)? 

I have a couple of dags which are very difficult to read using this new horizontal view and I can’t seem to find a setting or button to toggle the layout.

**Reply by braunjj** [reply]
Hey @toddy86, thanks for the feedback. We're looking into making this configurable via a toggle. Will follow up once we have a more concrete plan.

**Reply by toddy86** [reply]
Thanks for the reply. Ah that'd be great if there was a toggle. A toggle on a user and individual dag level would be ideal if that's possible :)

**Reply by toddy86** [reply]
@braunjj in the meantime, is there any way to revert back to the vertical always view? Presuming not by the sounds of it.
